src/path_segments.rs

```diff
@@ -18,13 +18,17 @@ use Url;
 /// Examples:
 ///
 /// ```rust
-/// # use url::Url;
-/// let mut url = Url::parse("mailto:me@example.com").unwrap();
+/// # use url::{Url, ParseError};
+///
+/// # fn run() -> Result<(), ParseError> {
+/// let mut url = Url::parse("mailto:me@example.com")?;
 /// assert!(url.path_segments_mut().is_err());
 ///
-/// let mut url = Url::parse("http://example.net/foo/index.html").unwrap();
+/// let mut url = Url::parse("http://example.net/foo/index.html")?;
 /// url.path_segments_mut().unwrap().pop().push("img").push("2/100%.png");
```

Comment:
That is something I would resolve in another PR: `path_segments_mut` returns `Result<_,()>`. With the API like this, we cannot capture it with the `?` without mapping the error or rolling a custom error type. Note that using `()` as error type is not recommended by the Rust API guidelines ([C-MEANINGFUL-ERR](https://github.com/brson/rust-api-guidelines#c-meaningful-err)).